src/System.Management.Automation/utils/Encoding.cs

```diff
@@ -151,7 +152,9 @@ public static Encoding GetEncoding(FileEncoding TextEncoding)
                     break;
 
                 case FileEncoding.BigEndianUTF32:
-                    result = Encoding.BigEndianUnicode;
+                    // This can possibly throw, but if so, we can't provide
+                    // the encoding which the user requested, so we should fail
+                    result = Encoding.GetEncoding("utf-32BE");
```

Comment:
Should we cache this in  internal static readonly constant as for `internal static readonly UTF8Encoding utf8NoBom`?